netdata-installer.sh

```diff
@@ -1121,7 +1121,7 @@ export NETDATA_CONFIGURE_OPTIONS="${NETDATA_CONFIGURE_OPTIONS}"
 INSTALL_UID="${UID}"
 if [ "\${INSTALL_UID}" != "\${UID}" ]
     then
-    echo >&2 "This script should be run as user with uid \${INSTALL_UID} but it now runs with uid \${UID}"
+    echo >&2 "You are running this script as user with uid \${UID}. The script should be run as root (user with uid \${INSTALL_UID})"
```

Comment:
Netdata can be installed as a user which has UID different than 0. In such case this message will be wrong.
Let's assume I am running installation as a typical user (UID 1000) and upgrade script as a user with UID 500, then this message would be:
> You are running this script as user with uid 500. The script should be run as root (user with uid 1000)

And that message is wrong as user root has ALWAYS uid = 0.

Reply:
Hi!

> Netdata can be installed as a user which has UID different than 0. In such case this message will be wrong.

OK, thats not clear from the message the user gets at this point.

Is there a definitive list of user/uids the update script can be run as? 

Put another way - is there a "best practice" user that should run netdata-updater.sh?

> You are running this script as user with uid 500. The script should be run as root (user with uid 1000)
> And that message is wrong as user root has ALWAYS uid = 0.

Yes, true, I missed the _INSTALL__. So potentially:

"You are running this script as user with uid \${UID}. Rerun the script as root (uid 0)."

Reply:
> Is there a definitive list of user/uids the update script can be run as?

There is no such list since we don't know what are the system settings. By default upgrade script is run as root since we link it to `/etc/cron.daily/netdata-updater`, but nothing is stopping people to put it into other user crontab and run from there.

That said, we are currently reworking our installation and upgrade system and this means that we will probably need to drop this feature and run upgrade script as root. It will be needed as we are moving to binary packages.

Upgrade path for upgrade script for upcoming months is like this:
1. Decouple it from `netdata-installer.sh` script (wip in #4558)
2. Drop usage of `git` and switch to downloading tarball artifacts from google cloud storage. Yesterday we merged a PR which builds uploads those daily
3. Work on binary packaging
4. Remove non-root auto-upgrade option
5. Switch upgrade script to use binary packages instead of tarballs

Reply:
> There is no such list since we don't know what are the system settings. By default upgrade script is run as root since we link it to /etc/cron.daily/netdata-updater, but nothing is stopping people from putting it into other user crontab and run from there.

OK, so the advisable user the script should be run as is root? Correct?

As a modification, - 

`echo >&2 "You are running this script as user with uid \${UID}. We recommend to run this script as root (user with uid 0)"
`
In this case, the (more experienced) user is still free to run the script as any user they want, and the less experienced user is helped to understand what they need to do.

Thoughts? I'm trying to help the less experienced user.

Reply:
Looks good to me